### PR TITLE
Fix query detection

### PR DESCRIPTION
**Pull request by katevoss**
Final fix to #1350

### CODE REVIEW

File: src/components/Router.js [hunk 1]
```diff
@@ -75,7 +76,7 @@ export default class Router extends Component {
       ({ pathname, search }) => this.setState({ pathname, search })
     )
 
-    if (isFirecloud() && new URLSearchParams(window.location.search).has('fcredir')) {
+    if (isFirecloud() && _.has('fcredir', qs.parse(Nav.history.location.search, { ignoreQueryPrefix: true, plainObjects: true }))) {
```
Comment (davidangb) [inline src/components/Router.js]: is this backwards? should it be `_.has(qs.parse(...), 'fcredir')`?

Comment (katevoss) [reply src/components/Router.js]: We use the `fp` version of lodash, which switches the arguments.

Comment (davidangb) [reply src/components/Router.js]: ah, gotcha, thanks!

**Review by davidangb — APPROVED** [review]
:+1: works for me; you may want a thumb from someone more familiar with the codebase.

This now only shows the notification when the query is after the hash. As Brian said, you might to check for both before and after, a  la:
```
qs.parse(window.location.search || Nav.history.location.search ...
```
(totally untested ^)

File: src/components/Router.js (hunk 1 above)
Comment (breilly2) [inline src/components/Router.js]: TOL: `URLSearchParams` might work just as well with `Nav.history.location` as it does with `window.location.search`... depending on how it handles the presence/absence of a query prefix. That said, it's probably better to standardize on `qs.parse` if that's what we're using. There will soon be a 3rd invocation of `qs.parse` in the code from my Fence PR, which may mean that it's time for some clean-up.

One way to clean it up is to have `Router` parse both `Nav.history.location.search` and `window.location.search` and pass both (or a union) down as props. Probably beyond the scope of this PR. Just looking for reactions to the idea here.

**Review by breilly2 — APPROVED** [review]
👍

File: src/components/Router.js (hunk 1 above)
Comment (zarsky-broad) [inline src/components/Router.js]: Can we do this:
```suggestion
    if (_.has('fcredir', qs.parse(Nav.history.location.search, { ignoreQueryPrefix: true, plainObjects: true }))) {
```
so it'll work right on non-prod?

**Pull request merged**